Auth0/Credentials.swift

```diff
@@ -28,24 +28,24 @@ import Foundation
 @objc(A0Credentials)
 public class Credentials: NSObject, JSONObjectPayload {
 
-    public let accessToken: String
-    public let tokenType: String
+    public let accessToken: String?
+    public let tokenType: String?
     public let idToken: String?
     public let refreshToken: String?
+    public var hasToken: Bool {
```

Comment:
Don't see the why we need this method, the `hasToken` can only be known to the one who starts the auth or if we pass along the `response_type` and validate. Thing we should do before yielding the credentials to the caller